### PR TITLE
sys/linux/epoll: add EPOLLWAKEUP constant, epoll_pwait() call and add documentation

### PR DESCRIPTION
**Pull request by ljmf00**
Signed-off-by: Luís Ferreira <contact@lsferreira.net>

### CODE REVIEW

File: src/core/sys/linux/epoll.d [hunk 1]
```diff
@@ -142,7 +145,84 @@ union epoll_data_t
...
+ * Returns: the number of triggered events returned in "events" buffer. Or -1 in
+ *          case of error with the "errno" variable set to the specific error
+ *          code.
+ */
 int epoll_wait (int epfd, epoll_event *events, int maxevents, int timeout);
+
+/**
+ * Wait for events on an epoll instance
+ *
+ * This function is a cancellation point and therefore not marked with __THROW.
```
Comment (ibuclaw) [inline src/core/sys/linux/epoll.d:212]: Are C-specific comments necessary?

Comment (ljmf00) [reply src/core/sys/linux/epoll.d:214]: This, in fact, generates Ddoc documentation. I ported C comments to D compatible documentation to make it easier to use the bindings by looking at the druntime reference.

Attached is an example of generated documentation for `epoll_pwait` routine:

image

**Comment by ljmf00** [comment]
I force pushed to fix a minor issue with documentation for enums. See: https://github.com/dlang/druntime/pull/3368/files#diff-e03064d1094968c0e0ccc96138a741a32070c6c34ef949fdaf47567a9f4b01d2L62-R69

File: src/core/sys/linux/epoll.d (hunk 1 above)
Comment (ibuclaw) [reply src/core/sys/linux/epoll.d:214]: Well, D doesn't have `__THROW`. :-)

File: src/core/sys/linux/epoll.d [hunk 2]
```diff
@@ -142,7 +147,84 @@ union epoll_data_t
...
+ *
+ * Params:
+ *   epfd = an epoll file descriptor instance
+ *   events = a buffer that will contain triggered events
+ *   maxevents = the maximum number of events to be returned ( usually size of
+ *               "events" )
+ *   timeout = specifies the maximum wait time in milliseconds (-1 == infinite)
+ *   ss = a signal set. May be specified as NULL, in which case epoll_pwait() is
```
Comment (ibuclaw) [inline src/core/sys/linux/epoll.d:222]: Likewise, D doesn't have `NULL`.  Though you could use `` `null` `` instead.

Comment (ljmf00) [reply src/core/sys/linux/epoll.d]: Done

File: src/core/sys/linux/epoll.d (hunk 1 above)
Comment (ljmf00) [reply src/core/sys/linux/epoll.d]: Oh yeah, got it. Fixed :)

**Review by RazvanN7 — APPROVED** [review]

**Pull request merged**